src/IndexManager.php

```diff
@@ -66,6 +66,8 @@ public function index($entities, ObjectManager $objectManager)
                     $objectManager->getClassMetadata($className),
                     $this->getNormalizer($className)
                 );
+
+                $objectManager->detach($entity);
```

Comment:
Don't ever do that unless **you're in control of the entity**. This listener definitely isn't and should never ever clear the object manager or detach single entities.

Reply:
You're right, we should do the detatch only when we are indexing from the CLI. In this context we're in control of the entity.

Reply:
Also, you should not detach individual entities but rather process data in batches and clear the entire UoW in between. Detaching selected entities leads to incomplete entity graphs which can really mess up your day.

Reply:
But isn't the point to allow GC collection asap after processing an individual entity ? I mean, I think we don't care about icomplete entity graph if it happens because if we scope the `detach` to only the indexation command, we're _sure_ that the detached entities won't be used after that. no ?

Reply:
Unless you can say that's 100% sure for every use-case of ORM and ODM out there, you shouldn't. Doing it any other way can also lead to performance issues, e.g. because the detach cascades to relationships causing the related entity to be refetched from the database.

Reply:
Mhhh. so it seems like a false good idea to use `detach` then;

Reply:
Correct.